src/control/cmd/daos/pool.go

```diff
@@ -329,6 +330,58 @@ func queryPool(poolHdl C.daos_handle_t, queryMask daos.PoolQueryMask) (*daos.Poo
 		if queryMask.HasOption(daos.PoolQueryOptionDisabledEngines) {
 			poolInfo.DisabledRanks = rs
 		}
+		if queryMask.HasOption(daos.PoolQueryOptionSuspectEngines) {
+			poolInfo.SuspectRanks = rs
+		}
+	}
+
+	return poolInfo, nil
+}
+
+func queryPool(poolHdl C.daos_handle_t, queryMask daos.PoolQueryMask) (*daos.PoolInfo, error) {
+	origQueryMask := queryMask
+	queryMask.ClearOptions(daos.PoolQueryOptionEnabledEngines,
+		daos.PoolQueryOptionSuspectEngines, daos.PoolQueryOptionDisabledEngines)
+	if origQueryMask.HasOption(daos.PoolQueryOptionEnabledEngines) {
+		queryMask.SetOptions(daos.PoolQueryOptionEnabledEngines)
+		origQueryMask.ClearOptions(daos.PoolQueryOptionEnabledEngines)
+	} else if origQueryMask.HasOption(daos.PoolQueryOptionDisabledEngines) {
+		queryMask.SetOptions(daos.PoolQueryOptionDisabledEngines)
+		origQueryMask.ClearOptions(daos.PoolQueryOptionDisabledEngines)
+	} else if origQueryMask.HasOption(daos.PoolQueryOptionSuspectEngines) {
+		queryMask.SetOptions(daos.PoolQueryOptionSuspectEngines)
+		origQueryMask.ClearOptions(daos.PoolQueryOptionSuspectEngines)
+	}
+	poolInfo, err := queryPoolHelper(poolHdl, queryMask)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I don't really understand why this part is necessary given that we go through the sequence of all three flags below.

Reply:
It is possible that we don't have all three flags enabled at all.. Notice if we only have one flag enabled, after we run this query, the following query won't be executed since we cleared original flag.

Reply:
Ah, I see. Thanks for the explanation. I think refactoring this area to reduce repetition will probably help make things clearer.